docs/software-architecture-and-design.md

```diff
@@ -266,22 +252,26 @@ We use the [boost Coroutine2 library](https://www.boost.org/doc/libs/1_71_0/libs
 
 
 ## How Do We Use Coroutines?
+
+> [!IMPORTANT]  
+> We are currently in the process of moving away from using coroutines and transitioning to using [finite-state machines](#finite-state-machines) for all our STP logic.
+
 We use Coroutines to write our [strategy logic](#strategy). The "pause and resume" functionality of Coroutines makes it much easier to write [Plays](#plays).
 
 Specifically, we use Coroutines as a way to break down our strategy into "stages". Once a "stage" completes we generally don't want to re-evaluate it, and would rather commit to a decision and move on. Coroutines makes it much easier to write "stages" of strategy without requiring complex state machine logic to check what stage we are in, and it's easier for developers to see what the intended order of operations is (eg. "Line up to take the shot" -> "shoot").
 
 In the past, we had issues with our gameplay logic "committing" to decisions if we were near certain edge cases. This caused robots to behave oddly, and sometimes get significantly slowed down in "analysis paralysis". Coroutines solve this problem by allowing us to write "stages" that execute top-to-bottom in a function, and once we make a decision we commit to it and move on to the next stage.
 
 Here's a more specific example. In this example we are going to pretend to write a [Tactic](#tactic) that will pass the ball.
-```
+```cpp
 def executeStrategy(IntentCoroutine::push_type& yield, Pass pass) {
     do {
         yield(/* align the robot to make the pass */)
-    }while(current_time < pass.start_time);
+    } while(current_time < pass.start_time);
     
     do {
         yield(/* kick the ball at the pass location */)
-    }while(/* robot has not kicked the ball */)
+    } while(/* robot has not kicked the ball */)
```

Comment:
a bit weird for us to show example code that doesn't follow our style guidelines